### ISSUE TITLE
AttributeError: 'Stream' object has no attribute 'add_event_listener'

### ISSUE THREAD

**Issue by adaszko** [jorisschellekens/borb#31]
Hi 👋 .  I have a problematic PDF file that triggers the error in the title of this issue.  Here's a minimal reproducible example:

```Python
#!/usr/bin/env python3
from borb.pdf.pdf import PDF
from borb.toolkit.text.simple_text_extraction import SimpleTextExtraction
l = SimpleTextExtraction()
f = open('tmp.pdf', 'rb')
doc = PDF.loads(f, [l])
```

The file that triggers it: [tmp.pdf](https://github.com/jorisschellekens/borb/files/7261790/tmp.pdf)

The stack trace:
```
Traceback (most recent call last):
  File "/Users/adaszko/repos/prawo/./bug.py", line 6, in <module>
    doc = PDF.loads(f, [l])
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/pdf/pdf.py", line 49, in loads
    return ReadAnyObjectTransformer().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/any_object_transformer.py", line 93, in transform
    return super().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/transformer.py", line 120, in transform
    out = h.transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/reference/xref_transformer.py", line 104, in transform
    trailer = self.get_root_transformer().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/any_object_transformer.py", line 100, in transform
    return super().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/transformer.py", line 120, in transform
    out = h.transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/object/dictionary_transformer.py", line 46, in transform
    v = self.get_root_transformer().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/any_object_transformer.py", line 100, in transform
    return super().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/transformer.py", line 120, in transform
    out = h.transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/reference/reference_transformer.py", line 103, in transform
    transformed_referenced_object = self.get_root_transformer().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/any_object_transformer.py", line 100, in transform
    return super().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/transformer.py", line 120, in transform
    out = h.transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/page/root_dictionary_transformer.py", line 53, in transform
    transformed_root_dictionary = t.transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/object/dictionary_transformer.py", line 46, in transform
    v = self.get_root_transformer().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/any_object_transformer.py", line 100, in transform
    return super().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/transformer.py", line 120, in transform
    out = h.transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/reference/reference_transformer.py", line 103, in transform
    transformed_referenced_object = self.get_root_transformer().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/any_object_transformer.py", line 100, in transform
    return super().transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/transformer.py", line 120, in transform
    out = h.transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/metadata/xmp_metadata_transformer.py", line 68, in transform
    out_value = super(XMPMetadataTransformer, self).transform(
  File "/Users/adaszko/repos/prawo/__pypackages__/3.9/lib/borb/io/read/object/stream_transformer.py", line 45, in transform
    object_to_transform.add_event_listener(l)  # type: ignore [attr-defined]
AttributeError: 'Stream' object has no attribute 'add_event_listener'
```

**Comment by jorisschellekens** [comment]
Hi there,

That's a result of a refactor that seems to have gone wrong a bit.
I removed the `EventListener` concept from PDF objects themselves (rather the process of reading/writing a PDF can now have `EventListener(s)` attached to it.

The method `add_event_listener` used to exist for any (internal) PDF object. This method was added dynamically (so that I could re-use some existing Python objects like `Decimal` or `str`).

Because my type-checker would always complain about the method not being defined (since the method is only known at runtime), I added explicit formatted comments to tell the type-checker to ignore any line that invokes the method.

That's probably why I didn't get a warning when I removed that method.

I'll make sure it gets fixed in the upcoming release (which should be this weekend).

Kind regards,
Joris Schellekens

**Comment by jorisschellekens** [comment]
Hi again,

The problem is fixed on my local version of the repo.

Would you like me to add an extra test to the repository (using your example document as input)?

Kind regards,
Joris Schellekens

**Comment by jorisschellekens** [comment]
Fixed in `v2.0.11`.
Verified against your document.

**Comment by adaszko** [comment]
Hi.  Sorry for a late reply.  I had few things going on.  Thank you for fixing this.  As for whether to include this as a test case, I'd say it's up to maintainers :)
